src/uikit/PillInputWithButton.js

```diff
@@ -12,40 +12,50 @@ const enhance = compose(
   withState('selected', 'setSelected', ({ defaultSelected }) => defaultSelected),
 );
 
-const PillInputWithButton = ({
+const SelectStyle = `
+  border-radius: 10px;
+  border-right: none;
+  border-top-right-radius: unset;
+  border-bottom-right-radius: unset;
+  flex-grow: 1;
+  height: 30px;
+`;
+
+export const PillInputWithButton = ({
   selected,
   setSelected,
   options,
   defaultSelected,
   children,
   onClick,
   render,
+  SelectComponent,
+  onOptionSelect,
+  isButtonDisabled,
 }) => {
   return (
     <LoadingOnClick
-      onClick={() => options[selected]()}
+      onClick={() => (onOptionSelect ? onOptionSelect({ selected }) : options[selected]())}
```

Comment:
it'd be nice to see this as the prop default, and always pass it `({ selected })`...
```
  ...
  SelectComponent,
  onOptionSelect = options[selected]
  ...
```